[Unified Search] `EuiSuperDatePicker` improvements

acstll opened the pull request:
## Summary

This enables 2 new features in the time picker `EuiSuperDatePicker` in Unified Search.

- Time window buttons: on the top-level of the component that allow shifting the time window backwards or forward, and zooming out.
- Time zone information displayed in the date/time picker popover.

<img width="1362" height="968" alt="image (4)" src="https://github.com/user-attachments/assets/846f307e-12eb-41b5-9eb7-35b3540a202f" />

## Changes

- Show time window buttons aa6e065ceacab0b1d4cf8db6cfbab1551de495ac
- Show time zone information aa6e065ceacab0b1d4cf8db6cfbab1551de495ac

For more info, see https://github.com/elastic/eui/pull/9210

acstll committed:
[Unified Search] Show time window buttons and time zone

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
Cloud deployment initiated, see credentials at: https://buildkite.com/elastic/kibana-deploy-cloud-from-pr/builds/560

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
PR Cloud deployment started at: https://buildkite.com/elastic/kibana-deploy-cloud-from-pr/builds/573

kibanamachine commented:
<!-- kbn-message-context:cloud-deploy-job -->
PR Cloud deployment started at: https://buildkite.com/elastic/kibana-deploy-cloud-from-pr/builds/574

acstll commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into eui/super_date_picker_batch_1_test

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/369518)
* Commit: 9dde9b23d2a1a3e49ae359e0ed65dae8b9c3b623

### Failed CI Steps
* [Build and Deploy to Cloud](https://buildkite.com/elastic/kibana-pull-request/builds/369518#019aeeca-f394-46ec-95c0-c702de250bf7)

### History
* :broken_heart: [Build #366409](https://buildkite.com/elastic/kibana-pull-request/builds/366409) failed aa6e065ceacab0b1d4cf8db6cfbab1551de495ac
* :broken_heart: [Build #366386](https://buildkite.com/elastic/kibana-pull-request/builds/366386) failed 09ec4ea244c71652379551bcd7a55694ac4fb479
* :broken_heart: [Build #361722](https://buildkite.com/elastic/kibana-pull-request/builds/361722) failed aa68d13d386135d6881ada561ebc37e41507354a
* :broken_heart: [Build #361410](https://buildkite.com/elastic/kibana-pull-request/builds/361410) failed 711a2aeba94f9feaca56415c02cd2a2326d58082
* :broken_heart: [Build #361366](https://buildkite.com/elastic/kibana-pull-request/builds/361366) failed bed4f1b033d0c450c9d77d767879ed0ac9f6f995

cc @acstll

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/369518","number":369518,"commit":"9dde9b23d2a1a3e49ae359e0ed65dae8b9c3b623"},{"buildStatus":{"state":"passed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366409","number":366409,"commit":"aa6e065ceacab0b1d4cf8db6cfbab1551de495ac"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/366386","number":366386,"commit":"09ec4ea244c71652379551bcd7a55694ac4fb479"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361722","number":361722,"commit":"aa68d13d386135d6881ada561ebc37e41507354a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361410","number":361410,"commit":"711a2aeba94f9feaca56415c02cd2a2326d58082"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/361366","number":361366,"commit":"bed4f1b033d0c450c9d77d767879ed0ac9f6f995"}],"number":369518}
buildkite-pr-comment-->

nreese reviewed (approved):
kibana-presentation changes LGTM
code review, tested in chrome

elasticmachine committed:
Merge branch 'main' into eui/super_date_picker_batch_1_test

acstll committed:
[FTR][Dashboards] Update baselines

QueryBarTopRow is taking wrapping in 2 lines at an earlier breakpoint because of the time window buttons, it used to wrap at ~970px now it does at ~1070px

acstll committed:
[FTP][Discover] Fix tooltip blocking click

I could not figure out the reason why a tooltip (Add filter) got stuck because of the change in layout (showing time window buttons)

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

akowalska622 reviewed (approved):
Data Discovery changes LGTM
